File: Level-1/48.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int cnt = 0;
        int zero_cnt = 0;
        for(int i = 0; i < lottos.length; i++){
            if(lottos[i] == 0) zero_cnt += 1;
            for(int j = 0; j < win_nums.length; j++){
                if(lottos[i] == win_nums[j]) cnt += 1;
            }
        }
        answer[0] = cnt + zero_cnt;
        answer[1] = cnt;
        for(int i = 0; i < answer.length; i++){
            switch(answer[i]){
                case 6 : answer[i] = 1; break;
                case 5 : answer[i] = 2; break;
                case 4 : answer[i] = 3; break;
                case 3 : answer[i] = 4; break;
                case 2 : answer[i] = 5; break;
                default : answer[i] = 6; break;
            }
        }
        return answer;
    }
}

File: Level-1/48.js
function solution(lottos, win_nums) {
    const rank = {
        0 : 6,
        1 : 6,
        2 : 5,
        3 : 4,
        4 : 3,
        5 : 2,
        6 : 1,
    };
    
    lottos.sort((a, b) => a - b);
    win_nums.sort((a, b) => a - b);
    
    let num_cnt = 0, zero_cnt = 0;
    lottos.forEach((lotto) => {
        win_nums.forEach((win_num) => {lotto === win_num ? num_cnt++ : ''});
        lotto === 0 ? zero_cnt++ : '';
    });
    
    return [rank[num_cnt + zero_cnt], rank[num_cnt]];
}
